app/services/katello/organization_creator.rb

```diff
@@ -0,0 +1,108 @@
+module Katello
+  class OrganizationCreator
+    attr_reader :library_view, :library_environment, :library_cvv, :content_view_environment, :anonymous_provider, :redhat_provider
+
+    def self.seed_all_organizations
+      User.as_anonymous_admin do
+        Organization.not_created_in_katello.each do |org|
+          self.new(org).seed!
+        end
+      end
+    end
+
+    def self.create_all_organizations!
+      if Katello::Ping.ping!(services: [:candlepin])
+        User.as_anonymous_admin do
+          Organization.not_created_in_katello.each do |org|
+            creator = self.new(org)
+            creator.create!
+          end
+        end
+      end
+    end
+
+    def initialize(organization)
+      @organization = organization
+    end
+
+    def seed!
+      @organization.setup_label_from_name
+      @organization.save!
+
+      create_library_environment
+      create_library_view
+      create_library_cvv
+      create_content_view_environment
+      create_anonymous_provider
+      create_redhat_provider
+    end
+
+    def create!
+      seed!
+
+      if needs_candlepin_organization?
+        ::Katello::Resources::Candlepin::Owner.create(@organization.label, @organization.name)
+      end
+
+      ::Katello::ContentViewManager.create_candlepin_environment(
+        content_view_environment: @content_view_environment
+      )
+
+      @organization.created_in_katello = true
+      @organization.save!
```

Comment:
may be nice to add an audit comment here conveying that the Katello-ey-ness of the organization was consummated!

Reply:
I'm wondering what the value of that is. I view Audits as the record of things that happened which users care about and this is more of an internal process, not something they need to track or monitor.

Reply:
Maybe it would be better as a log then -- I was just thinking about what if something failed halfway through `seed` and the creation was left incomplete?  How could we make that easier to discover and address?

Reply:
Keeping in mind that the OrganizationCreator should be resilient to failures at any step (failed runs will not impede subsequent ones), here are the scenarios:

the db:seed process fails => This will raise an error just like it would today.

org create via API, UI fails => Error will be raised as it would today

middleware for org creation fails => it pings candlepin before starting the process and if that ping fails then a UINotification is sent. any other error will be raised. that should likely change (since right now it will basically make the app unusable..) but I'm having trouble envisioning where this could break down such that an error gets raised in creating the handful of entities required for a fully-fledged org

Reply:
Take a look at latest commit - it's not fully baked but I think it captures what you have in mind. I need to compare the current behavior to understand implications of removing the created candlepin environment and owner if there were any problems.